google/cloud/aiplatform/jobs.py

```diff
@@ -2485,7 +2485,7 @@ def update(
             current_job.model_deployment_monitoring_objective_configs = (
                 ModelDeploymentMonitoringJob._parse_configs(
                     objective_configs,
-                    current_job.endpoint,
+                    aiplatform.Endpoint(current_job.endpoint),
                     deployed_model_ids,
                 )
             )
```

Comment:
This call seems to have no effect if the job isn't running but doesn't raise or warn in that case. Also this silent noop behavior isn't documented. This should be changed because it seems like the update is executed successfully when no update is actually performed.

Reply:
Service side will throw an error if the job cannot be updated (i.e. when it's not in RUNNING stage). It doesn't fail silently.

Reply:
It seems like the update is only attempted when RUNNING:

https://github.com/googleapis/python-aiplatform/blob/9a506eeae6be3fa33b262dc7782d0bb4f10bf765/google/cloud/aiplatform/jobs.py#L2492-L2497

Reply:
Oh wait you're right. I have removed the if conditions in the update/pause/delete functions to avoid silently masking failed actions. I also added system test coverage for failed update when job is in PENDING state.